content/guides/nodejs/containerize.md

```diff
@@ -60,228 +74,806 @@ This utility will walk you through creating the following files with sensible de
+  db:
+    image: postgres:16-alpine
+    container_name: todoapp-db
+    environment:
+      POSTGRES_DB: '${POSTGRES_DB:-todoapp}'
+      POSTGRES_USER: '${POSTGRES_USER:-todoapp}'
+      POSTGRES_PASSWORD: '${POSTGRES_PASSWORD:-todoapp_password}'
+    volumes:
+      - postgres_data:/var/lib/postgresql/data
+    ports:
+      - '${DB_PORT:-5432}:5432'
+    restart: unless-stopped
+    healthcheck:
+      test: ['CMD-SHELL', 'pg_isready -U ${POSTGRES_USER:-todoapp} -d ${POSTGRES_DB:-todoapp}']
+      interval: 10s
+      timeout: 5s
+      retries: 5
+      start_period: 5s
+    networks:
+      - todoapp-network
+
+# ========================================
+# Volume Configuration
+# ========================================
+volumes:
+  postgres_data:
+    name: todoapp-postgres-data
+    driver: local
+
+# ========================================
+# Network Configuration
+# ========================================
+networks:
+  todoapp-network:
+    name: todoapp-network
+    driver: bridge
+```
+
+This Docker Compose configuration includes:
+
+- **Development service** (`app-dev`): Full development environment with hot reload, debugging support, and bind mounts
+- **Production service** (`app-prod`): Optimized production deployment with resource limits and security hardening
+- **Database service** (`db`): PostgreSQL 16 with persistent storage and health checks
+- **Networking**: Isolated network for secure service communication
+- **Volumes**: Persistent storage for database data
+
+## Create environment configuration
+
+Create a `.env` file to configure your application settings:
+
+```console
+$ cp .env.example .env
+```
+
+Update the `.env` file with your preferred settings:
+
+```env
+# Application Configuration
+NODE_ENV=development
+APP_PORT=3000
+VITE_PORT=5173
+DEBUG_PORT=9229
+
+# Production Configuration
+PROD_PORT=8080
+PROD_MEMORY_LIMIT=2G
+PROD_CPU_LIMIT=1.0
+PROD_MEMORY_RESERVATION=512M
+PROD_CPU_RESERVATION=0.25
+
+# Database Configuration
+POSTGRES_HOST=db
+POSTGRES_PORT=5432
+POSTGRES_DB=todoapp
+POSTGRES_USER=todoapp
+POSTGRES_PASSWORD=todoapp_password
+DB_PORT=5432
+
+# Security Configuration
+ALLOWED_ORIGINS=http://localhost:3000,http://localhost:5173
+```
+
+---
 
-# Comments are provided throughout this file to help you get started.
-# If you need more help, visit the Dockerfile reference guide at
-# https://docs.docker.com/go/dockerfile-reference/
+## Build the Docker image
 
-# Want to help us make this template better? Share your feedback here: https://forms.gle/ybq9Krt8jtBL3iCk7
+The default Dockerfile generated by `docker init` provides a reliable baseline for standard Node.js applications. However, since this project is a full-stack TypeScript application that includes both a backend API and frontend React components, the Dockerfile should be customized to better support and optimize this specific architecture.
 
-ARG NODE_VERSION=18.0.0
+### Step 1: Review the generated files
```

Comment:
This section seems like an intro to the rest of the steps rather than a step it's self.
```suggestion
```